Add checklist to docs handbook (#2350)
kubernetes/sig-release: release-team/role-handbooks/docs/Release-Timeline.md

### Attend meetings

⚠️  Attend the release team meeting outlined in the current release. If you cannot attend, a Shadow needs to attend in your place.
⚠️  Attend [SIG Docs meetings](https://github.com/kubernetes/community/tree/master/sig-docs#meetings) to provide a status update. 

> If you cannot attend, a Shadow needs to attend in your place or update the [SIG Docs meeting notes](https://docs.google.com/document/d/1emuO4nmaQq3K8JZ9-MQeIygtrCPO9kWv7U7RzTaW4F8/edit?usp=sharing) with an update.

Reviewer: ```suggestion
If you cannot attend, a Shadow needs to attend in your place and/or update the [SIG Docs meeting notes](https://docs.google.com/document/d/1emuO4nmaQq3K8JZ9-MQeIygtrCPO9kWv7U7RzTaW4F8/edit?usp=sharing) with the information necessary to keep the community informed (e.g. staffing updates, branch syncs, placeholder PR deadlines, etc).
```

Author: added suggestion and squashed

- Here is a checklist to guide the weekly status report. 

> _Note:_ SIG Docs prefers and invites a status of yellow or red on anything that needs attention or possiblity to miss deadlines or target dates.

Reviewer: ```suggestion
_Note:_ SIG Docs prefers and welcomes a status of yellow or red on anything that needs attention, has the possiblity to miss deadlines or target dates, or requires follow-up from feature authors and SIGs that you're not receiving.
```

Author: added suggestion and squashed

[ ] Is the integration branch healthy (does not need rebase)?

> [ ] Has the weekly branch sync been done?

Reviewer: ```suggestion
[ ] Has the weekly branch sync been done? (a lapse in 1-2 branch syncs indicates 'yellow' status, 3 or more lapses should be deemed red)
```

Author: added suggestion and squashed

> [ ] Has the PR for the weekly branch sync merged or needs review?

Reviewer 2: I would also add here that if there are merge conflicts that they need help with, it is a matter of concern that needs to be flagged appropriately as well. It would be worthwhile to explore the option of the Release docs lead/their shadows sending out a short message on the #release-docs channel like the ci-signal team does on the mailing lists.

Author: modified this to line to read
>[ ] Has the PR for the weekly branch sync merged or needs review? If assistance is needed to resolve merge conflicts, please flag others by communicating on the #release-docs Slack channel

> [ ] A week before the Docs PR Placeholder deadline, have 80% of the enhancements that opted in for docs created a placeholder PR?

Reviewer: ```suggestion
[ ] A week before the Docs PR Placeholder deadline, have 80% of the enhancements that opted in for docs created a placeholder PR? (anything below 60% opt-in for docs where placeholder PRs are needed should indicate yellow (or red) status, contingent on the amount)
```

Author: added suggestion and squashed

[ ] Three days before the Docs PR Placeholder deadline, have 90% of the enhancements that opted in for docs created a placeholder PR?

> [ ] A week before the Docs Ready for Review deadline, are 80% of the docs PRs ready for review?

Reviewer: ```suggestion
[ ] A week before the Docs Ready for Review deadline, are 80% of the docs PRs ready for review? (anything below 60% should indicate yellow (or red) status, contingent on the amount)
```

Author: added suggestion and squashed

[ ] Three days before the Docs Ready for Review deadline, are 90% of the docs PRs ready for review?

> [ ] A week before the Docs Complete deadline, have 80% of the docs PRs merged?

Reviewer: ```suggestion
[ ] A week before the Docs Complete deadline, have 80% of the docs PRs merged? (anything below 60% should indicate yellow (or red) status, contingent on the amount)
```

Author: added suggestion and squashed

> [ ] Three days before the Docs Complete deadline, have 90% of the docs PRs merged?

Reviewer: ```suggestion
[ ] Three days before the Docs Complete deadline, have 90% of the docs PRs merged? (be sure to mention if you need the help of SIG Docs approvers/reviewers to meet your deadline)
```

Author: added suggestion and squashed
